### DISCUSSION TITLE
Star colors

### DISCUSSION THREAD

**Discussion by altoidaddict** [PlummersSoftwareLLC/NightDriverStrip#97] [Q&A]
Either there is a code issue or I am missing something in the expectation of how rainbow colors are intended to function in the star effects, none of the following actually yield any rainbow colors, just red and white:
new StarryNightEffect<MusicStar>("Rainbow Music Stars", RainbowColors_p, 2.0, 2, LINEARBLEND, 5.0, 0.0, 10.0), // Rainbow Music Star

new StarryNightEffect<BubblyStar>("Little Blooming Rainbow Stars", BlueColors_p, prob, 4, LINEARBLEND, 2.0, 0.0, mult), // Blooming 
Little Rainbow Stars

new StarryNightEffect<BubblyStar>("Big Blooming Rainbow Stars", RainbowColors_p, 2, 12, LINEARBLEND, 1.0), // Blooming Rainbow Stars

Am I doing something wrong?  I decided to do a fresh clone from git earlier today and can't seem to get anything other than red and white.

**Comment by Username-cannot-be-blank** [comment]
I'm having the same issue. It looks like the break happens when the Star constructor is called in stareffect.h . It calls MovingFadingPaletteObject without passing the color index, so when MovingFadingPaletteObject calls FadingPaletteObject, it defaults to 0 for the palette index, which is the first color in the array. At least, I think that's what's going on.
